File: Ems_backend/src/main/java/com/ak/ems/service/DepartmentService.java
package com.ak.ems.service;

import com.ak.ems.dto.DepartmentDto;
import com.ak.ems.response.PageResponse;



public interface DepartmentService {
    DepartmentDto createDepartment(DepartmentDto departmentDto);
    DepartmentDto getDepartmentById(Long departmentId);
    PageResponse<DepartmentDto> getAllDepartments(String query, int page, int size, String sortBy, String sortDir);
    DepartmentDto updateDepartment(Long departmentId, DepartmentDto departmentDto);
    void deleteDepartment(Long departmentId);
}

File: Ems_frontend/src/services/DepartmentService.js
import apiClient from "../apiConfig";

const BASE_URL = "/departments";

export const getAllDepartments = () => apiClient.get(BASE_URL);
export const createDepartment = (department) => apiClient.post(BASE_URL, department);
export const getDepartmentById = (id) => apiClient.get(`${BASE_URL}/${id}`);
export const updateDepartment = (id, department) => apiClient.put(`${BASE_URL}/${id}`, department);
export const deleteDepartment = (id) => apiClient.delete(`${BASE_URL}/${id}`);
